services/systeminfo/service/src/main/java/org/collectionspace/services/systeminfo/SystemInfoResource.java

```diff
@@ -31,74 +28,87 @@
-		// TODO Auto-generated method stub
-		return null;
-	}
+    private static final int MAJOR_IDX = 0;
+    private static final int MINOR_IDX = 1;
+    private static final int PATCH_IDX = 2;
 
-	@Override
-	public String getServiceName() {
-		return SystemInfoClient.SERVICE_NAME;
-	}
+    @Override
+    public Class<?> getCommonPartClass() {
+        // TODO Auto-generated method stub
+        return null;
+    }
 
-	@Override
-	protected String getVersionString() {
-		// TODO Auto-generated method stub
-		return null;
-	}
+    @Override
+    public String getServiceName() {
+        return SystemInfoClient.SERVICE_NAME;
+    }
+
+    @Override
+    protected String getVersionString() {
+        // TODO Auto-generated method stub
+        return null;
+    }
 
-	//
-	// API Endpoints
-	//
+    //
+    // API Endpoints
+    //
 
     @GET
     public SystemInfoCommon get(@Context UriInfo ui) {
-    	SystemInfoCommon result = null;
-
-    	try {
-    		result = new SystemInfoCommon();
-    		result.setInstanceId(ServiceMain.getInstance().getCspaceInstanceId());
-    		result.setDisplayName(ServiceMain.VER_DISPLAY_NAME);
-    		Version ver = new Version();
-    		ver.setMajor(ServiceMain.VER_MAJOR);
-    		ver.setMinor(ServiceMain.VER_MINOR);
-    		ver.setPatch(ServiceMain.VER_PATCH);
-    		ver.setBuild(ServiceMain.VER_BUILD);
-    		result.setVersion(ver);
-
-    		result.setHostTimezone(TimeZone.getDefault().getID());
-    		result.setHostLocale(Locale.getDefault().toLanguageTag());
-    		result.setHostCharset(Charset.defaultCharset().name());
-    		//
-    		// To get the full set of the system information, we required the user be authenticated *and* have "DELETE" privs on the "systeminfo" resource
-    		//
-    		try {
-    			ServiceContext<SystemInfoCommon, SystemInfoCommon> ctx = createServiceContext(getServiceName(), ui);
-    			CSpaceResource res = new URIResourceImpl(ctx.getTenantId(), SystemInfoClient.SERVICE_NAME, HttpMethod.DELETE);
-    			if (AuthZ.get().isAccessAllowed(res)) {
-							// TODO: Stop hardcoding this!
-							// result.setNuxeoVersionString("7.10-HF17");
-							result.setHost(String.format("Architecture:%s Name:%s Version:%s",
-									System.getProperty("os.arch"), System.getProperty("os.name"), System.getProperty("os.version")));
-							result.setJavaVersionString(System.getProperty("java.version"));
-							// TODO: Stop hardcoding this!
-							// result.setPostgresVersionString("9.5.7");
-					}
-				} catch (UnauthorizedException e) {
-    				logger.trace(e.getMessage(), e);
-    		}
-
-    	} catch(Exception e) {
-    		Response response = Response.status(Response.Status.BAD_REQUEST).entity(e.getMessage()).type("text/plain").build();
+        SystemInfoCommon result;
+
+        try {
+            result = new SystemInfoCommon();
+            result.setInstanceId(ServiceMain.getInstance().getCspaceInstanceId());
+            result.setDisplayName(ServiceMain.VER_DISPLAY_NAME);
```

Comment:
`ServiceMain.VER_DISPLAY_NAME` currently contains a hardcoded version number. I think we should change that to be a template string, and fill in the major and minor numbers here with `String.format`.